### PR TITLE
Replace record summary cards with an identifier bar

### PR DESCRIPTION
**Pull request by Bonapara**
## Summary

- Replace record summary cards with a compact identifier bar and integrate it with tabbed pages and merge previews.
- Align the bar with the pinned column; preserve tab overflow and recovery after resizing.
- Keep the existing pinning behavior. This layer adds no persisted pinning field or Unpin action.

## Screenshots

Captured from this PR’s preview using seeded demo data. The before image comes from the preceding PR, #24935, at the same 1176 × 720 viewport.

### Record page: summary card → compact identifier bar

The compact identity stays separate from the navbar and aligns with the pinned field column.

| Before (#24935) | After (#24936) |
| --- | --- |
| Before: Google record with the large summary card | After: Google record with the compact identifier bar |

<details>
<summary>Pinned-tab settings and responsive tab overflow</summary>

**Edit the pinned tab.** Home remains accessible in edit mode. Pinning behavior is unchanged; the Unpin action is introduced in #24937.

Pinned Home tab settings in edit mode

**Keep overflow tabs accessible.** Opening the settings panel narrows the content area and moves remaining tabs into More. Closing it was checked to restore the full tab list.

More menu containing the tabs that no longer fit

</details>

<details>
<summary>Merge preview and mobile layout</summary>

**Merge preview.** The same compact identifier appears above the previewed fields. No records were merged.

Merge preview with the compact Google identifier

**Mobile.** Record identity, the Home tab, horizontally scrollable tabs, and fields remain accessible at 390 × 640.

Mobile record page with compact identity and Home tab

</details>

## Stack

Part 2 of 3 splitting #24860. Review and merge in order: #24935 → #24936 → #24937. Base: `bonapara/shared-header-identifiers`.

Review the layout, edit-mode pinned-tab access, responsive surfaces, and tab sizing separately from optional-pinning persistence.

### CODE REVIEW

File: packages/twenty-front/src/modules/page-layout/components/__stories__/PageLayoutTabList.stories.tsx [hunk 1]
```diff
@@ -173,3 +203,31 @@ export const Default: Story = {
     isReorderEnabled: true,
   },
 };
+
+export const IdentifierBar: Story = {
+  args: {
+    presentation: 'identifier-bar',
+    isReorderEnabled: false,
```
[the remark this answers is not in the document]
Comment (Bonapara) [reply packages/twenty-front/src/modules/page-layout/components/__stories__/PageLayoutTabList.stories.tsx:249]: Fixed in a953086d51. All four identifier-bar stories now have play functions, and the story supplies the required layout-rendering context. Wide/centered variants verify tab selection; narrow variants select a hidden tab through More and verify a newly added tab remains reachable. All five tab-list stories passed in the built-in browser.

File: packages/twenty-front/src/modules/page-layout/components/__tests__/PageLayoutRecordIdentifierBar.test.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+    expect(screen.queryByText('Tasks')).not.toBeInTheDocument();
+
+    await user.click(
+      screen.getByRole('button', { name: 'Edit pinned tab: Tasks' }),
+    );
+
+    expect(mockOpenTabSettings).toHaveBeenCalledWith('pinned-tab-id');
+
+    rerender(
```
[the remark this answers is not in the document]
Comment (Bonapara) [reply packages/twenty-front/src/modules/page-layout/components/__tests__/PageLayoutRecordIdentifierBar.test.tsx:111]: Fixed in a953086d51. Added toHaveBeenCalledTimes(1) alongside the existing argument assertion, so a duplicated settings-open callback fails the test. The focused identifier-bar, tab measurement, and overflow utility suites passed (23 tests).

File: packages/twenty-front/src/modules/ui/layout/tab-list/hooks/__tests__/useTabListMeasurements.test.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    hook.result.current.onContainerWidthChange({
+      width: containerWidth,
+      height: 48,
+    });
+  });
+
+  return hook;
+};
+
+describe('useTabListMeasurements', () => {
```
[the remark this answers is not in the document]
Comment (Bonapara) [reply packages/twenty-front/src/modules/ui/layout/tab-list/hooks/__tests__/useTabListMeasurements.test.ts:46]: Fixed in a953086d51. Moved the arithmetic and right-padding boundary cases into calculateVisibleTabCount.test.ts. The hook suite now covers only reactive changes: resize, changed padding, add-button removal, measuring a new tab, and tab removal. The utility and hook tests passed, as did the narrow-layout browser interactions.

File: packages/twenty-front/src/modules/page-layout/components/PageLayoutSingleTabRenderer.tsx [hunk 4]
```diff
@@ -49,11 +62,12 @@ const PageLayoutSingleTabRendererInner = () => {
 
   return (
     <>
-      <SummaryCard
-        objectNameSingular={targetRecordIdentifier.targetObjectNameSingular}
-        objectRecordId={targetRecordIdentifier.id}
-        isInSidePanel={isInSidePanel}
-      />
+      <StyledRecordIdentifierBar>
+        <RecordIdentifierBarTitle
```
Comment (Weiko) [inline packages/twenty-front/src/modules/page-layout/components/PageLayoutSingleTabRenderer.tsx:69]: We lose the creation date here in both merge surfaces. `SummaryCard` displayed it and the regular identifier bar still does. Can we also render `RecordIdentifierBarCreatedAt` here?

**Review by Weiko — APPROVED** [review]
See comment before merging and check if it needs to be addressed.
LGTM for the rest 👍

File: packages/twenty-front/src/modules/page-layout/components/PageLayoutSingleTabRenderer.tsx (hunk 4 above)
Comment (Bonapara) [reply packages/twenty-front/src/modules/page-layout/components/PageLayoutSingleTabRenderer.tsx:69]: Fixed in e5d9bca82c. Both source-record and merged-preview headers now reuse `RecordIdentifierBarCreatedAt`. The regression test checks the date changes with the selected record and stays absent when `createdAt` is missing. After rebasing onto main, 48 focused tests, frontend typechecking, and diff lint/format checks pass. Verified the Merge preview, First, and Second tabs in the browser as well.
